formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java

```diff
@@ -1068,7 +1069,51 @@ public void export(final HttpServerRequest request) {
     @ResourceFilter(ShareAndOwner.class)
     @SecuredAction(value = MANAGER_RESOURCE_RIGHT, type = ActionType.RESOURCE)
     public void shareJson(final HttpServerRequest request) {
-        super.shareJson(request, false);
+        final String id = request.params().get("id");
+        if (id != null && !id.trim().isEmpty()) {
+            UserUtils.getUserInfos(this.eb, request, user -> {
+                if (user == null) {
+                    String message = "[Formulaire@shareJson] User not found in session.";
+                    log.error(message);
+                    unauthorized(request, message);
+                    return;
+                }
+
+                super.shareService.shareInfos(user.getUserId(), id, I18n.acceptLanguage(request), request.params().get("search"), shareInfosEvt -> {
+                    if (shareInfosEvt.isLeft()) {
+                        log.error("[Formulaire@shareJson] Fail to get sharing infos");
+                        renderInternalError(request, shareInfosEvt);
+                        return;
+                    }
+
+                    JsonObject shareInfos = shareInfosEvt.right().getValue().copy();
+                    formService.get(id, user, formEvt -> {
+                        if (formEvt.isLeft()) {
+                            log.error("[Formulaire@shareJson] Fail to get form with id " + id);
+                            renderInternalError(request, formEvt);
+                            return;
+                        }
+
+                        JsonObject form = formEvt.right().getValue();
+                        if (form.getBoolean("is_public")) {
```

Comment:
https://rules.sonarsource.com/java/RSPEC-5411

```suggestion
if (Boolean.TRUE.equals(form.getBoolean("is_public", false)))
```

Je pense rajouter false en 2eme param pour éviter un potentiel NPE dans ta condition (en plus de la règle condition)

et magic string "is_public"